client/src/modules/shipment/create-shipment.js

```diff
@@ -182,11 +176,11 @@ function CreateShipmentController(
     vm.depot = depot;
     vm.stockForm.setDepot(depot);
 
-    // set the shipment origin
     vm.shipment.origin_depot_uuid = vm.depot.uuid;
 
-    // Delete the old destination depot
-    delete vm.shipment.destination_depot_uuid;
+    if (vm.isCreateState || (vm.shipment.destination_depot.uuid === depot.uuid)) {
```

Comment:
I'm not sure I understand this logic.  Can you leave a comment to explain it?

Reply:
Done.  Note that I plan to add an issue to prevent changing the origin depot on a shipment.   Then this code should be revisited and cleaned up.